### PR TITLE
handle s3 timeout on settings page

### PR DESCRIPTION
**Pull request by ruslandoga**
### Changes

This PR reduces export-fetch request attempts to one and handles the resulting error (if there is one). This allows import feature to continue working when S3 (used in exports) is not available.

Screenshot 2024-04-23 at 23 06 22

### Tests
- [x] Automated tests have been added

### Changelog
- [x] This PR does not make a user-facing change

### Documentation
- [x] This change does not need a documentation update

### Dark mode
- [x] The UI has been tested both in dark and light mode

### CODE REVIEW

File: lib/plausible_web/live/csv_export.ex [hunk 1]
```diff
@@ -33,76 +33,83 @@ defmodule PlausibleWeb.Live.CSVExport do
...
-        "executing" ->
-          # Exports.oban_notify/1 is called in `perform/1` and
-          # the notification arrives while the job.state is still "executing"
-          if export = get_export.(site_id) do
-            assign(socket, status: "ready", export: export)
-          else
-            assign(socket, status: "in_progress")
+    %{storage: storage, site: site, site_id: site_id} = socket.assigns
+
+    assign_async(socket, [:status, :export], fn ->
```
Comment (ruslandoga) [inline lib/plausible_web/live/csv_export.ex:38]: I went with a quick-fix using `assign_async`. Alternative approach is reducing the number of attempts in `Exports.get_s3_export/1` to 1 and adding one more `:status` for `export_fetch_failure`. Any preference?

Comment (zoldar) [reply lib/plausible_web/live/csv_export.ex:38]: It's fine with `assign_async` 👍  Perhaps we could extract `get_export` assingment and `Exports.get_last_export_job/1` outside of it? Not a big deal though.

**Review by zoldar — APPROVED** [review]

File: test/plausible_web/controllers/site_controller_test.exs [hunk 2]
```diff
@@ -744,6 +744,32 @@ defmodule PlausibleWeb.SiteControllerTest do
...
+
+    setup tags do
+      if tags[:async] do
+        raise "this test modifies application envinronment and can't be run asynchronously"
+      end
+
+      prev_env = Application.get_env(:ex_aws, :s3)
+      new_env = Keyword.update!(prev_env, :port, fn prev_port -> prev_port + 1 end)
+      Application.put_env(:ex_aws, :s3, new_env)
+      on_exit(fn -> Application.put_env(:ex_aws, :s3, prev_env) end)
```
Comment (ruslandoga) [inline test/plausible_web/controllers/site_controller_test.exs:765]: `patch_env` only supports `:plausible` app

File: test/plausible_web/controllers/site_controller_test.exs [hunk 3]
```diff
@@ -744,6 +744,32 @@ defmodule PlausibleWeb.SiteControllerTest do
...
+    end
+
+    @tag :capture_log
+    test "displays error message", %{conn: conn, site: site} do
```
Comment (ruslandoga) [inline test/plausible_web/controllers/site_controller_test.exs:769]: I'll need to skip this test in CE.

Comment (ruslandoga) [reply test/plausible_web/controllers/site_controller_test.exs:769]: https://github.com/plausible/analytics/pull/4036/commits/f83581e276fb6879760ef2e295d4016edf5190c4

**Review by zoldar — APPROVED** [review]

**Pull request merged**